controller/gateway/controller.go

```diff
@@ -704,6 +724,10 @@ func dataplaneSpecDeepEqual(spec1, spec2 *operatorv1beta1.DataPlaneOptions) bool
 		return false
 	}
 
+	if !reflect.DeepEqual(spec1.Extensions, spec2.Extensions) {
+		return false
+	}
+
 	return true
```

Comment:
nit:
```suggestion
	return reflect.DeepEqual(spec1.Extensions, spec2.Extensions) {
```
won't be the same?